core/src/validation.jl

```diff
@@ -406,59 +406,82 @@ function sorted_table!(
     return table
 end
 
+struct NodeID
+    value::Int
+end
+
+Base.broadcastable(id::NodeID) = Ref(id)
+Base.show(io::IO, id::NodeID) = print(io, "#$(id.value)")
+
+function Base.isless(id_1::NodeID, id_2::NodeID)::Bool
+    return id_1.value < id_2.value
+end
+
+function Base.getindex(M::AbstractArray, id_row::NodeID, id_col::NodeID)
+    return M[id_row.value, id_col.value]
+end
+
+function Base.setindex!(
+    M::AbstractArray,
+    value::T,
+    id_row::NodeID,
+    id_col::NodeID,
+)::Nothing where {T}
+    M[id_row.value, id_col.value] = value
+    return nothing
+end
```

Comment:
These are only to support `flow::SparseMatrix` handling right, and therefore can be removed once we stop storing flow as a SparseMatrix? Perhaps good to write this in a comment above.